MapboxCoreNavigationTests/PassiveLocationDataSourceTests.swift

```diff
@@ -59,45 +60,52 @@ class PassiveLocationDataSourceTests: XCTestCase {
+        do {
+            try locationManager.configureNavigator(withTilesVersion: tilesVersion)
+        } catch {
+            XCTAssertTrue(false)
         }
-        wait(for: [startingExpectation], timeout: 2)
-        
-        // FIXME: Starting the location manager should automatically configure the navigator.
-        XCTAssertNoThrow(try locationManager.configureNavigator(withTilesVersion: "1234"))
-        
-        let locationUpdateExpectation = expectation(description: "Location manager should respond to every manual location update")
-        locationUpdateExpectation.expectedFulfillmentCount = 5
-        
+
+        let locationUpdateExpectation = expectation(description: "Location manager takes some time to start mapping locations to a road graph")
+        locationUpdateExpectation.expectedFulfillmentCount = 1
+
         let road = Road(from: CLLocationCoordinate2D(latitude: 47.207966, longitude: 9.527012), to: CLLocationCoordinate2D(latitude: 47.209518, longitude: 9.522167))
         let delegate = Delegate(road: road, locationUpdateExpectation: locationUpdateExpectation)
+        locationManager.updateLocation(CLLocation(latitude: 47.208674, longitude: 9.524650))
         locationManager.delegate = delegate
-        
-        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(10)) {
-            locationManager.updateLocation(CLLocation(latitude: 47.208674, longitude: 9.524650))
-        }
-        
-        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(25)) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
             locationManager.updateLocation(CLLocation(latitude: 47.208943, longitude: 9.524707))
-            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
-                locationManager.updateLocation(CLLocation(latitude: 47.209082, longitude: 9.524319))
-                DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
-                    locationManager.updateLocation(CLLocation(latitude: 47.209229, longitude: 9.523838))
-                    DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
-                        locationManager.updateLocation(CLLocation(latitude: 47.209612, longitude: 9.522629))
-                        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
-                            locationManager.updateLocation(CLLocation(latitude: 47.209842, longitude: 9.522377))
-                        }
-                    }
-                }
-            }
+            locationManager.updateLocation(CLLocation(latitude: 47.209082, longitude: 9.524319))
+            locationManager.updateLocation(CLLocation(latitude: 47.209229, longitude: 9.523838))
+            locationManager.updateLocation(CLLocation(latitude: 47.209612, longitude: 9.522629))
+            locationManager.updateLocation(CLLocation(latitude: 47.209842, longitude: 9.522377))
```

Comment:
Each of these location updates has more or less the same timestamp. Would that throw off the navigator and make this an unrealistic test? We could set these CLLocations’ timestamps to be more offset from each other, which would at least come closer to the asynchronous test we had before.